File: BondPrototype/Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace BondPrototype.Models;
// aaaa
public class Movie
{
    [Key] public int Id { get; set; }
    public string Title { get; set; }
    public byte Rating { get; set; }
    public DateTime ReleaseDate { get; set; }
    public string MoviePoster { get; set; }
    public string Description { get; set; }
    
    public Person DirectedBy { get; set; }
    public List<Person> Actors { get; set; }
    public List<Award> Awards { get; set; }
}

File: BondPrototype/Angular/src/app/api/models/movie.ts
import {Person} from './person';
import {Award} from './award';
import {key} from '../actions/decorators';

export class Movie {
    @key() public id: number;
    public title: string;
    public rating: number;
    public releaseDate: Date;
    public moviePoster: string;
    public description: string;
    public directedBy: Person;
    public actors: Person[];
    public awards: Award[];
}